controller/seafile-controller.c

```diff
@@ -820,6 +821,33 @@ read_seafdav_config()
         ctl->seafdav_config.fastcgi = FALSE;
     }
 
+    /* host */
+    gchar* host = g_key_file_get_string(key_file, "WEBDAV", "host", &error);
+    unsigned char host_valid = 1;
+    if (error != NULL) {
+        if (error->code != G_KEY_FILE_ERROR_KEY_NOT_FOUND) {
+            seaf_message ("Error when reading WEBDAV.host, use default value 'localhost'\n");
+        }
+	host_valid = 0;
+    }else{
+	// no error occured while reading host, validate it
+        size_t len = strnlen(host,SEAFDAV_MAX_HOST + 1);
+        if(len ==  0 || len > SEAFDAV_MAX_HOST){
+	    seaf_message ("Error when validating WEBDAV.host, use default value 'localhost'\n");
+	    host_valid = 0;
+	}
+    }
+    if(host_valid){
+	// copy entered host to seafdav settings
+	strncpy(ctl->seafdav_config.host,host,SEAFDAV_MAX_HOST);
+    }else{
+	// use default 'localhost'
+	strcpy(ctl->seafdav_config.host,"localhost");
+    }
+    if(host){
+     	free(host);
```

Comment:
use `g_free` instead of `free` here, since the string `host` is returned from `g_key_file_get_string`.